sql-plugin/src/main/scala/com/nvidia/spark/rapids/NvtxRangeWithDoc.scala

```diff
@@ -38,7 +38,7 @@ object NvtxRegistry {
 
   private def register(id: NvtxId): Unit = {
     if (registeredRanges.contains(id.name)) {
-      Console.err.println(s"Collision detected for key: ${id.name}")
+      throw new IllegalArgumentException(s"Collision detected for key: ${id.name}")
     } else {
       registeredRanges += (id.name -> id)
     }
```

Comment:
nit, no need to change if no other edits required: more concisely and efficiently it can be done along the lines
```Scala
val prev = registeredRanges.put(id.name, id)
require(prev.isEmpty, s"Collision detected for key: ${id.name}")
```

Reply:
Good to know, thanks!